google/cloud/security/common/data_access/dao.py

```diff
@@ -84,8 +96,16 @@ def load_data(self, resource_name, timestamp, data):
         """
         with csv_writer.write_csv(resource_name, data) as csv_file:
             try:
-                snapshot_table_name = self._create_snapshot_table(
-                    resource_name, timestamp)
+                try:
```

Comment:
Can you move table creation to a new function? I think that would make this more readable.